conda-store-server/conda_store_server/_internal/server/views/api.py

```diff
@@ -15,45 +15,27 @@
 from conda_store_server._internal import orm, schema
 from conda_store_server._internal.environment import filter_environments
 from conda_store_server._internal.server import dependencies
+from conda_store_server._internal.server.pagination import (
+    Cursor,
+    CursorPaginatedArgs,
+    OrderingMetadata,
+    paginate,
+)
 from conda_store_server.conda_store import CondaStore
 from conda_store_server.exception import CondaStoreError
 from conda_store_server.server import schema as auth_schema
 from conda_store_server.server.auth import Authentication
 from conda_store_server.server.schema import AuthenticationToken, Permissions
 
-
-class PaginatedArgs(TypedDict):
-    """Dictionary type holding information about paginated requests."""
-
-    limit: int
-    offset: int
-    sort_by: List[str]
-    order: str
-
-
-router_api = APIRouter(
+router_api_v1 = APIRouter(
     tags=["api"],
     prefix="/api/v1",
 )
 
-
-def get_paginated_args(
-    page: int = 1,
-    order: Optional[str] = None,
-    size: Optional[int] = None,
-    sort_by: List[str] = Query([]),
-    server=Depends(dependencies.get_server),
-) -> PaginatedArgs:
-    if size is None:
-        size = server.max_page_size
-    size = min(size, server.max_page_size)
-    offset = (page - 1) * size
-    return {
-        "limit": size,
-        "offset": offset,
-        "sort_by": sort_by,
-        "order": order,
-    }
+router_api_v2 = APIRouter(
```

Comment:
Can the v2 router and associated endpoints go in it's own module (maybe like `api_v2.py`??). I think that'll make it easier to understand that there are multiple living versions of the api and make it easier to make changes in the future.